tools/envoy/fetch.sh

```diff
@@ -15,26 +16,33 @@ source "$(dirname -- "${BASH_SOURCE[0]}")/../common.sh"
 
 function download_envoy() {
     local binary_name=$1
+    local bin_dir=$(dirname ${BINARY_PATH})
     echo "Downloading ${binary_name}"
 
     if [ ! -d "$(dirname "${BINARY_PATH}")" ]; then
       mkdir -p "$(dirname "${BINARY_PATH}")"
     fi
 
+    local tar_path="${BINARY_PATH}.tar.gz"
+
     local status
-    status=$(curl -# --location --output "${BINARY_PATH}" --write-out '%{http_code}' \
-    "https://download.konghq.com/mesh-alpine/${binary_name}")
+    status=$(curl -# --location --output "${tar_path}" --write-out '%{http_code}' \
+    "https://github.com/kumahq/envoy-builds/releases/download/${ENVOY_TAG}/${binary_name}.tar.gz")
 
     if [ "$status" -ne "200" ]; then
         rm "${BINARY_PATH}"
         msg_err "Error: failed downloading Envoy: ${status} error"
     fi
 
+    tar -C "$bin_dir" -xzvf "${tar_path}"
+    rm $tar_path
+    mv "${bin_dir}/envoy-${ENVOY_DISTRO}" ${BINARY_PATH}
```

Comment:
github doesn't allow me to add the suggestion for the whole section, but maybe we could instead do something like:
```shell
curl --location  \
  "https://github.com/kumahq/envoy-builds/releases/download/${ENVOY_TAG}/${binary_name}.tar.gz" \
  | tar --extract --gzip --verbose "envoy-${ENVOY_DISTRO}" --directory "${BINARY_PATH}"
```

Reply:
talked offline. It does not work on MacOS. We would need to do some conditions based on OS, so the current approach is simpler in the end.